### DISCUSSION TITLE
How to identify what's giving "decompiler parameter ID" trouble

### DISCUSSION THREAD

**Discussion by RenaKunisaki** [NationalSecurityAgency/ghidra#2871] [Q&A]
I have a program where, when I run the "Decompiler Parameter ID" analyzer, I can see it correctly deduce the parameters of most functions... then when it's nearly finished, it goes back and gives every function 16 parameters (8 GPRs, 8 FPRs). I assume there's some function it's tripping over that's confusing it. How can I find where the trouble is?

**Comment by astrelsky** [comment]
Is there a varargs function being called? Sometimes it struggles with printf and can't differentiate between a parameter and a local. If this is the case try overriding the varargs function signature.
